pilot/pkg/model/push_context.go

```diff
@@ -843,11 +838,12 @@ func (ps *PushContext) initSidecarScopes(env *Environment) error {
 	// build sidecar scopes for other namespaces that dont have a sidecar CRD object.
 	// Derive the sidecar scope from the root namespace's sidecar object if present. Else fallback
 	// to the default Istio behavior mimicked by the DefaultSidecarScopeForNamespace function.
-	for _, s := range ps.ServiceByHostname {
-		ns := s.Attributes.Namespace
-		if len(ps.sidecarsByNamespace[ns]) == 0 {
-			// use the contents from the root namespace or the default if there is no root namespace
-			ps.sidecarsByNamespace[ns] = []*SidecarScope{ConvertToSidecarScope(ps, rootNSConfig, ns)}
+	for _, nsMap := range ps.ServiceByHostnameAndNamespace {
```

Comment:
Please add a few comments. 

Any change in behavior ? My understanding from reading the code is  that if there is no explicit Sidecar, this will make services visible only in the namespace where they are defined - is it correct ?